core/state.go

```diff
@@ -657,7 +667,7 @@ func (s *State) purgeContract(addr *felt.Felt) error {
 	return storageCloser()
 }
 
-func (s *State) buildReverseDiff(blockNumber uint64, diff *StateDiff) (*StateDiff, error) {
```

Comment:
are you sure that logic left the same?

Reply:
Yup, pretty sure the logic is left the same. I just extracted the deletion logic to a separate function, and they range over the same `blockNumber` and `stateDiff`, so everything should be the exact same (it's a lot clearer when comparing the changes in VSCode).